### PR TITLE
gh-113804: Add 'x' format to float.format()

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-113804

----
📚 Documentation preview 📚: https://cpython-previews--119945.org.readthedocs.build/

### CODE REVIEW

**Comment by vstinner** [comment]
Not supported:
* Precision
* ``'='`` alignment

**Comment by vstinner** [comment]
cc @mdickinson @skirpichev @serhiy-storchaka

File: Doc/library/string.rst [hunk 1]
```diff
@@ -571,6 +571,12 @@ The available presentation types for :class:`float` and
    |         | the current locale setting to insert the appropriate     |
    |         | number separator characters.                             |
    +---------+----------------------------------------------------------+
+   | ``'x'`` | For :class:`float`, hex format. Outputs the number in    |
+   |         | base 16, using lower-case letters for the digits above 9.|
+   |         | The alternate form adds ``'0x'`` prefix.                 |
```
Comment (skirpichev) [inline Doc/library/string.rst:576]:
```suggestion
   | ``'x'`` | For :class:`float`, hex format, supported by             |
   |         | :meth:`~float.fromhex`.   Represent the number by a      |
   |         | hexadecimal string in the form ``[±][0x]h[.hhh]p±d``,    |
   |         | using using lower-case letters for the digits above 9.   |
   |         | The exponent ``d`` is written in decimal, it always      |
   |         | contains at least one digit, and it gives the power of 2 |
   |         | by which to multiply the coefficient.                    |
   |         |                                                          |
   |         | The alternate form adds ``'0x'`` prefix.                 |
```

1) maybe the first sentence + remark about lower-case is enough
2) maybe we should document also 1-normalisation & case of subnormals.

File: Doc/library/string.rst [hunk 2]
```diff
@@ -571,6 +571,12 @@ The available presentation types for :class:`float` and
...
+   |         |                                                          |
+   |         | Precision and ``'='`` alignment are not supported.       |
```
Comment (skirpichev) [inline Doc/library/string.rst]: Hmm, what's wrong with ``=``?  This seems to be supported on my branch:
```pycon
>>> f"{-1.1:30x}"
'           -1.199999999999ap+0'
>>> f"{-1.1:=30x}"
'-           1.199999999999ap+0'
>>> f"{-1.111111:=30f}"
'-                     1.111111'
>>> f"{-1.111111:30f}"
'                     -1.111111'
```

**Comment by serhiy-storchaka** [comment]
Why not simply use `float.hex()`?

**Comment by vstinner** [comment]
> Why not simply use float.hex()?

Do you mean: instead of adding this format? This format gives more formatting options: sign, 0x prefix, alignment.

**Comment by serhiy-storchaka** [comment]
It is not so difficult to process the result of `float.hex()`. And I think that this is a pretty rare case, it is not worth the cost of added code.

**Comment by vstinner** [comment]
@mdickinson @tim-one: What do you think of this approach? Do you agree with @serhiy-storchaka that the existing `float.hex()` method is enough?

File: Doc/library/string.rst (hunk 2 above)
Comment (vstinner) [reply Doc/library/string.rst]: I was just too lazy to implement it. If it matters, I can change my implementation to support the "=" alignment as well.

**Comment by mdickinson** [comment]
@vstinner For me having `x` support for float formatting would be no more than a minor convenience - I'd use it primarily for the padding, alignment and sign-handling options. But yes, I could do all that with custom functionality around `float.hex`. So call me +0 on this.

I do think `=` should be supported, if we implement this.

**Comment by vstinner** [comment]
@mdickinson:
> But yes, I could do all that with custom functionality around float.hex. So call me +0 on this. I do think = should be supported, if we implement this.

Ok, I rewrote my PR to implement the ``"="`` alignment (and simplify the implementation).

**Comment by vstinner** [comment]
It was decided to not pursue this approach, since float.hex() exists and it's enough for most usages, and we don't want to make str.format() more complicated.

**Pull request closed without merging**